Dockerfile

```diff
@@ -1,18 +1,64 @@
+ARG RUBY_VERSION=3.2.2
 # Use the official Ruby image as the parent image
-FROM ruby:3.2.2
+FROM registry.docker.com/library/ruby:$RUBY_VERSION-slim as base
 
 # Set the working directory to /app
 WORKDIR /app
 
+# Set production environment this values will be overwritten at with docker-compose.yml
+ARG BUNDLE_WITHOUT
+ENV BUNDLE_DEPLOYMENT="1" \
+    BUNDLE_PATH="/usr/local/bundle" \
+    RAILS_LOG_TO_STDOUT="1" \
+    RAILS_SERVE_STATIC_FILES="true" \
+    BUNDLE_WITHOUT="${BUNDLE_WITHOUT}"
+
+# Throw-away build stage to reduce size of final image
+FROM base as build
+
+# Install libvips for Active Storage preview support
+RUN apt-get update -qq && \
+    apt-get install --no-install-recommends -y \
+    build-essential \
+    git \
+    pkg-config \
+    libpq-dev \
+    postgresql-client \
+    nodejs
+
 # Copy the Gemfile and Gemfile.lock into the image
 COPY Gemfile Gemfile.lock ./
 
 # Install dependencies
-RUN bundle install
+RUN bundle install && \
+    rm -rf ~/.bundle/ "${BUNDLE_PATH}"/ruby/*/cache "${BUNDLE_PATH}"/ruby/*/bundler/gems/*/.git && \
+    bundle exec bootsnap precompile --gemfile
 
 # Copy the rest of the application code into the image
 COPY . .
 
+# Precompile bootsnap code for faster boot times
+RUN bundle exec bootsnap precompile app/ lib/
+
+# To ignore precompile step we can send ENV['RUNNING_CI']=true to dockerfile
+# RSpotify.authenticate in /config/application.rb brakes the build if spotify ENV vars not set right
+ARG RUNNING_CI="false"
+# Precompiling assets for container without requiring secret RAILS_MASTER_KEY
+RUN if [ "$RUNNING_CI" = "false" ]; then \
+      SECRET_KEY_BASE_DUMMY=1 bin/rails assets:precompile && bundle exec rake tailwindcss:build; \
+    fi
```

Comment:
This will pass the failed build due to `RSpotify.authenticate` fail on precompile stage

Reply:
I didn't get why `RSpotify.authenticate` run in precompile stage? and if we know that it will be fail, could we skip this step by default to set `RUNNING_CI=false`? @erimicel